src/effects/builtin/noisegen/noisegenerator.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include "PerTrackEffect.h"
+#include "SettingsVisitor.h"
+#include "common/generatoreffect.h"
+
+namespace au::effects {
+struct NoiseSettings
+{
+    enum Type
+    {
+        White,
+        Pink,
+        Brownian,
+        Count
+    };
+
+    static constexpr Type typeDefault = Type::White;
+    static constexpr double amplitudeDefault = 0.8;
+
+    Type type { typeDefault };
+    double amplitude { amplitudeDefault };
+};
+
+class BUILTIN_EFFECTS_API NoiseGenerator : public GeneratorEffect, public EffectWithSettings <NoiseSettings, PerTrackEffect>
```

Comment:
minor: the `BUILTIN_EFFECTS_API` macros and alike aren't necessary in AU4, where libraries are linked statically.